### ISSUE TITLE
aws_dms_endpoint mongo settings default case

### ISSUE THREAD

**Issue by jhmartin** [terraform-providers/terraform-provider-aws#8672]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

```
Terraform v0.11.14
+ provider.aws v2.10.0
```

### Affected Resource(s)

* aws_dms_endpoint

### Terraform Configuration Files

```hcl
resource "aws_dms_endpoint" "mongodb" {
  endpoint_id   = "mongo-source"
  endpoint_type = "source"
  engine_name   = "mongodb"
  database_name = "example"
  server_name   = "mongo.example.com"
  port          = 27017
  username      = "dummy"
  password      = "dummy"

  mongodb_settings {
    auth_mechanism = "default"
  }

}
```

### Expected Behavior

Configuration converges

### Actual Behavior

After initial creation, all subsequent plans include:
```
  ~ aws_dms_endpoint.mongodb
      mongodb_settings.0.auth_type:     "password" => "PASSWORD"
      mongodb_settings.0.nesting_level: "none" => "NONE"
```

Explicitly setting those values as lowercase results in an empty plan:
```
    auth_type = "password"
    nesting_level = "none"
```

### Steps to Reproduce

1. `terraform plan`

### References

Looks like https://github.com/terraform-providers/terraform-provider-aws/blob/6309aee2bb8668a0f3aaa82c5f34ba49088c1907/aws/resource_aws_dms_endpoint.go#L141 and https://github.com/terraform-providers/terraform-provider-aws/blob/6309aee2bb8668a0f3aaa82c5f34ba49088c1907/aws/resource_aws_dms_endpoint.go#L151 should be lowercase.  

https://docs.aws.amazon.com/dms/latest/APIReference/API_MongoDbSettings.html shows conflicting information about the case of the value but the behavior I see is it is implicitly downcased.

**Comment by nywilken** [comment]
@jhmartin sorry your running into trouble here. Looking at the schema and the AWS SDK should be updated to use the provided SDK constants https://godoc.org/github.com/aws/aws-sdk-go/service/databasemigrationservice#pkg-constants

Which contain the proper casing for each of the default values expected by the API. Thanks for reporting.

**Comment by richardjennings** [comment]
I have created a PR https://github.com/terraform-providers/terraform-provider-aws/pull/8795

**Comment by bflad** [comment]
The fix for this has been merged and will release with version 2.13.0 of the Terraform AWS Provider, likely later this week. 👍

**Comment by bflad** [comment]
This has been released in [version 2.13.0 of the Terraform AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#2130-may-31-2019). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
